src/pages/on-chain-governance/_components/InputComponent.tsx

```diff
@@ -26,6 +27,15 @@ export function InputComponent({
   isVisited,
   error,
 }: InputComponentProps): JSX.Element {
+  const ref = useRef<HTMLTextAreaElement>(null);
+  const dimension = useWindowDimensions();
+  useEffect(() => {
+    if (ref.current) {
+      ref.current.style.height = "24px";
+      ref.current.style.height = `${ref.current.scrollHeight}px`;
```

Comment:
why are we assigning to the LHS twice?

Reply:
line 35 is for expanding to the scroll height then line 34 is for minimizing the textbox size

Reply:
dont get it, wouldn't line 35 overwrite line 34?

Reply:
```
// Set initial base height to get the correct scrollHeight
ref.current.style.height = "24px";
// Then set the height directly
ref.current.style.height = `${ref.current.scrollHeight}px`;
```

scrollHeight is dependent on the initial base height even though line 35 is overriding line34 it will use line34 to get the correct scrollHeight